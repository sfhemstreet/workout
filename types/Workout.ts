import { Difficulty } from "./Difficulty";
import { Exercise } from "./Exercise";
import { Tag } from "./Tag";

export type Workout = {
  id: string;
  name: string;
  description: string;
  exercises: Exercise[];
  rounds: number;
  difficulty: Difficulty;
  stars: number;
  creator: {
    id: string;
    name: string;
    avatar: string;
  }
  isShared: boolean;
  clonedFromIds: Workout["id"][];
  createdAt: Date;
  tags: Tag[]
};