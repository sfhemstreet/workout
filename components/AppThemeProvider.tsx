import { ThemeProvider } from "@emotion/react";
import { ReactNode } from "react";

import { darkColors, lightColors } from "../styles/colors";
import { GlobalStyles } from "../styles/GlobalStyles";
import { theme } from "../styles/theme";
import { ThemeType } from "../types/ThemeType";

type AppThemeProviderProps = {
  children: ReactNode;
  themeType: ThemeType;
};
/**
 * AppThemeProvider
 * - provides the current theme in use to child components.
 * - provides the current theme to Emotion styled components via Emotion's ThemeProvider.
 * - applies global CSS via GlobalStyles.
 *
 * @param children
 */
export const AppThemeProvider: React.FC<AppThemeProviderProps> = ({
  children,
  themeType,
}) => (
  <ThemeProvider
    theme={{
      ...theme,
      name: themeType,
      colors: themeType === "LIGHT" ? lightColors : darkColors,
    }}
  >
    <GlobalStyles themeType={themeType} />
    {children}
  </ThemeProvider>
);
