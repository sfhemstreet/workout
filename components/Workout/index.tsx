import styled from "@emotion/styled";
import React, { useEffect, useState } from "react";
import { TransitionStatus } from "react-transition-group";

import { REST_PERIOD } from "../../constants";
import { SurfaceElevation } from "../../styles/SurfaceElevation";
import { ActiveExercise, ActiveWorkout } from "../../types";
import { SecondaryButton } from "../Buttons";
import { InlineButton } from "../Buttons/InlineButton";
import { Confetti } from "./Confetti";
import { CurrentExerciseDetails } from "./CurrentExerciseDetails";
import { CurrentExerciseDisplay } from "./CurrentExerciseDisplay";
import { FadableDiv, FadeInOut, FadeInOutMixin } from "../FadeInOut";
import { FullscreenBlurredBackground } from "../FullScreenBlurredBackground";
import { NextExercisesList } from "./NextExercisesList";
import { RoundsDisplay } from "./RoundsDisplay";
import { H3, H4, NumberSpan } from "../Txt";

type WorkoutProps = {
  currentExercise: ActiveExercise;
  workout: ActiveWorkout;
  actions: {
    addWorkoutToList?: () => void;
    toggleStartStop: () => void;
    togglePause: () => void;
    previousExercise: () => void;
    nextExercise: () => void;
    goToWorkouts: () => void;
    restartWorkout: () => void;
  };
};

/**
 * Workout
 *
 * Displays UI that user completes a workout with.
 * Displays the workout name, current exercise, current time,
 * has button for displaying details modal, list of exercises,
 * and shoots confetti when completed.
 *
 * @param currentExercise
 * @param workout
 * @param actions
 */
export const Workout = ({
  currentExercise,
  workout,
  actions,
}: WorkoutProps) => {
  const [isFiringConfetti, setIsFiringConfetti] = useState(false);
  const [isDescriptionShowing, setIsDescriptionShowing] = useState(false);

  const handleToggleDescription = () => {
    if (
      workout.isStarted &&
      !currentExercise.isPaused &&
      currentExercise.duration > 0 &&
      !isDescriptionShowing
    ) {
      actions.togglePause();
    }

    setIsDescriptionShowing(!isDescriptionShowing);
  };

  const nextExerciseIndex = workout.exercises.reduce(
    (acc, curr, currIndex, self) =>
      currIndex + 1 < self.length && curr.id === workout.currentExerciseId
        ? currIndex + 1
        : acc,
    0
  );

  const nextExerciseName =
    workout.exercises.length > 0
      ? nextExerciseIndex === 0 && workout.currentRound === workout.rounds
        ? undefined
        : workout.exercises[nextExerciseIndex].name === REST_PERIOD
        ? "Rest"
        : workout.exercises[nextExerciseIndex].name
      : "";

  useEffect(() => {
    if (workout.id === "") return;

    if (workout.isCompleted) {
      setTimeout(() => setIsFiringConfetti(true), 300);
    } else if (isFiringConfetti) {
      setIsFiringConfetti(false);
    }
  }, [workout]);

  return (
    <Container>
      {workout.id !== "" ? (
        <>
          <Top>
            <Title>
              Workout:<H4 textAlign="center">{workout.name}</H4>
            </Title>
            <RoundsDisplay
              currentRound={workout.currentRound}
              totalRounds={workout.rounds}
              isStarted={workout.isStarted}
            />
          </Top>

          <FadeInOut
            isShowing={!workout.isCompleted}
            timeout={{ exit: 0 }}
            render={(transitionStatus) => (
              <Exercise transitionStatus={transitionStatus}>
                <CurrentExerciseDisplay
                  {...currentExercise}
                  isStarted={workout.isStarted}
                  isCompleted={workout.isCompleted}
                  onClickTitle={handleToggleDescription}
                  nextExercise={nextExerciseName}
                />
              </Exercise>
            )}
          />

          <FadeInOut
            isShowing={workout.isCompleted}
            timeout={{ enter: 300, appear: 300, exit: 0 }}
            render={(transitionStatus) => (
              <Completed transitionStatus={transitionStatus}>
                <H3 textAlign="center">
                  <NumberSpan>WORKOUT COMPLETED</NumberSpan>
                </H3>
                <H3 textAlign="center">Congratulations!</H3>
              </Completed>
            )}
          />

          <FadeInOut
            isShowing={!workout.isCompleted}
            timeout={{ exit: 0 }}
            render={(transitionStatus) => (
              <List transitionStatus={transitionStatus}>
                <NextExercisesList
                  exercises={workout.exercises}
                  currentExerciseId={workout.currentExerciseId}
                />
              </List>
            )}
          />

          <ActionsPanel>
            <FadeInOut
              isShowing={!workout.isStarted}
              timeout={{ exit: 0 }}
              render={(transitionStatus) => (
                <ExploreArea transitionStatus={transitionStatus}>
                  <InlineButton
                    onClick={(e) => {
                      e.preventDefault();
                      actions.goToWorkouts();
                    }}
                    isFullWidth
                    isCentered
                  >
                    Explore Workouts
                  </InlineButton>
                </ExploreArea>
              )}
            />

            <FadeInOut
              timeout={{ exit: 0 }}
              isShowing={workout.isStarted}
              render={(transitionStatus) => (
                <PreviousArea transitionStatus={transitionStatus}>
                  <SecondaryButton
                    onClick={(e) => {
                      e.preventDefault();
                      actions.previousExercise();
                    }}
                    disabled={
                      workout.exercises.length <= 1 ||
                      (workout.exercises.findIndex(
                        (e) => e.id === workout.currentExerciseId
                      ) <= 0 &&
                        workout.currentRound === 1)
                    }
                  >
                    Back
                  </SecondaryButton>
                </PreviousArea>
              )}
            />

            <FadeInOut
              timeout={{ exit: 0 }}
              isShowing={workout.isStarted}
              render={(transitionStatus) => (
                <NextArea transitionStatus={transitionStatus}>
                  <SecondaryButton
                    onClick={(e) => {
                      e.preventDefault();
                      actions.nextExercise();
                    }}
                  >
                    Next
                  </SecondaryButton>
                </NextArea>
              )}
            />

            <FadeInOut
              timeout={{ exit: 0 }}
              isShowing={workout.isStarted}
              render={(transitionStatus) => (
                <PauseArea transitionStatus={transitionStatus}>
                  <SecondaryButton
                    onClick={(e) => {
                      e.preventDefault();
                      actions.togglePause();
                    }}
                    disabled={currentExercise.duration <= 0}
                  >
                    {currentExercise.isPaused ? "Resume" : "Pause"}
                  </SecondaryButton>
                </PauseArea>
              )}
            />

            <StartQuitArea isStarted={workout.isStarted}>
              <StartQuitBtn
                isQuit={workout.isStarted}
                onClick={(e) => {
                  e.preventDefault();
                  workout.isCompleted
                    ? actions.restartWorkout()
                    : actions.toggleStartStop();
                }}
              >
                {workout.isStarted
                  ? "Quit"
                  : workout.isCompleted
                  ? "Restart"
                  : "Start"}
              </StartQuitBtn>
            </StartQuitArea>
          </ActionsPanel>

          {/* Description Modal */}
          {currentExercise.name !== REST_PERIOD && (
            <FadeInOut
              isShowing={isDescriptionShowing}
              render={(transitionStatus) => (
                <FullscreenBlurredBackground
                  transitionStatus={transitionStatus}
                  onClick={handleToggleDescription}
                >
                  <CurrentExerciseDetails
                    onClose={handleToggleDescription}
                    {...currentExercise}
                  />
                </FullscreenBlurredBackground>
              )}
            />
          )}

          <FadeInOut
            timeout={{ exit: 0 }}
            isShowing={typeof actions.addWorkoutToList !== "undefined"}
            render={(transitionStatus) => (
              <SecondaryButton
                onClick={(e) => {
                  e.preventDefault();
                  actions.addWorkoutToList && actions.addWorkoutToList();
                }}
              >
                Add workout to list
              </SecondaryButton>
            )}
          />
        </>
      ) : (
        <Title>You need to select a workout!</Title>
      )}

      {workout.isCompleted && <Confetti isFiring={isFiringConfetti} />}
    </Container>
  );
};

const Container = styled.div`
  padding: 0px 10px 10px 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;

  @media ${(p) => p.theme.media.tablet} {
    padding: 10px;
  }

  @media ${(p) => p.theme.media.laptop} {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 60px 1fr min-content;
    grid-gap: 10px;

    max-width: 1040px;
  }

  @media ${(p) => p.theme.media.laptopL} {
    //grid-gap: 20px;
  }
`;

const Top = styled.div`
  /* ${(p) => SurfaceElevation(p.theme.name, 1)}; */
  padding-left: 1rem;
  padding-bottom: 0.5rem;
  border-radius: 10px;

  @media ${(p) => p.theme.media.laptop} {
    grid-column: 1 / 3;
    grid-row: 1;
    justify-self: center;
  }
`;

const Title = styled.div`
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
`;

const Exercise = styled(FadableDiv)`
  @media ${(p) => p.theme.media.laptop} {
    grid-column: 2;
    grid-row: 2;
    justify-self: center;
  }
`;

const List = styled(FadableDiv)`
  @media ${(p) => p.theme.media.laptop} {
    grid-column: 1;
    grid-row: 2;
    justify-self: center;
    align-self: center;
  }
`;

const Completed = styled.div<{ transitionStatus: TransitionStatus }>`
  width: 300px;
  height: 450px;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  opacity: ${(p) => (p.transitionStatus === "entered" ? 1 : 0)};
  transition: all 800ms ease-in-out;

  @media ${(p) => p.theme.media.laptop} {
    grid-column: 1 / 3;
    grid-row: 2;
    justify-self: center;
    align-self: center;
  }
`;

const ActionsPanel = styled.ul`
  list-style: none;

  padding: 0px;
  padding-bottom: 30px;

  width: 300px;
  min-width: 300px;

  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr;
  grid-gap: 5px 5px;

  transition: ${(p) => p.theme.transitions.normal};

  li {
    //width: 83px;
    min-width: 83px;
    height: 36px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  @media ${(p) => p.theme.media.laptop} {
    grid-column: 1 / 3;
    grid-row: 3;
    justify-self: center;
    align-self: start;
    padding-bottom: 5px;
    margin-top: 0px;
  }
`;

const StartQuitArea = styled.li<{
  isStarted: boolean;
}>`
  grid-column: 2 / 3;
  grid-row: ${(p) => (p.isStarted ? 3 : 1)};
`;

const PreviousArea = styled.li<{ transitionStatus: TransitionStatus }>`
  ${(p) => FadeInOutMixin(p.transitionStatus)};
  grid-column: 1;
  grid-row: 2;
`;

const PauseArea = styled.li<{ transitionStatus: TransitionStatus }>`
  ${(p) => FadeInOutMixin(p.transitionStatus)};
  grid-column: 2 / 3;
  grid-row: 1;
`;

const NextArea = styled.li<{ transitionStatus: TransitionStatus }>`
  ${(p) => FadeInOutMixin(p.transitionStatus)};
  grid-column: 3;
  grid-row: 2;
`;

const ExploreArea = styled.li<{ transitionStatus: TransitionStatus }>`
  ${(p) => FadeInOutMixin(p.transitionStatus)};
  grid-column: 2 / 3;
  grid-row: 3;
`;

const StartQuitBtn = styled(SecondaryButton)<{ isQuit: boolean }>`
  display: flex;
  justify-content: center;
  align-items: center;

  :hover {
    ${(p) => p.isQuit && `color: ${p.theme.colors.error};`};
  }
`;
