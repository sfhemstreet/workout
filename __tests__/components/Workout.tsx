import { Workout } from "../../components/Workout";
import { render } from "../test-utils/test-util";

test("Workout renders properly", () => {
  const workout = render(
    <Workout
      currentExercise={{
        id: "1",
        name: "Exercise 1",
        description: "Test",
        isPaused: false,
        currentTime: 5,
        duration: 10,
        repetitions: 0,
      }}
      workout={{
        id: "w1",
        name: "Workout 1",
        description: "test workout",
        rounds: 3,
        currentRound: 2,
        currentExerciseId: "1",
        isStarted: true,
        isCompleted: false,
        difficulty: "hard",
        stars: 2,
        isShared: true,
        tags: ["arms", "core", "back"],
        clonedFromIds: [],
        createdAt: new Date(2021, 2, 2),
        creator: {
          id: "user1",
          name: "Test-User",
          avatar: "https://robohash.org/test-user",
        },
        exercises: [
          {
            id: "1",
            name: "Exercise 1",
            description: "Test d",
            duration: 10,
            repetitions: 0,
          },
          {
            id: "2",
            name: "Exercise 2",
            description: "Test d2",
            duration: 20,
            repetitions: 0,
          },
          {
            id: "3",
            name: "Exercise 3",
            description: "Test d3",
            duration: 0,
            repetitions: 30,
          },
        ],
      }}
      actions={{
        toggleStartStop: () => {},
        togglePause: () => {},
        previousExercise: () => {},
        nextExercise: () => {},
        goToWorkouts: () => {},
      }}
    />
  );

  expect(workout).toMatchSnapshot();
});

test("Workout with Confetti renders properly", () => {
  const workout = render(
    <Workout
      currentExercise={{
        id: "1",
        name: "Exercise 1",
        description: "Test",
        isPaused: false,
        currentTime: 5,
        duration: 10,
        repetitions: 0,
      }}
      workout={{
        id: "w1",
        name: "Workout 1",
        description: "test workout",
        rounds: 3,
        currentRound: 2,
        currentExerciseId: "1",
        isStarted: true,
        isCompleted: true,
        difficulty: "hard",
        stars: 2,
        isShared: true,
        tags: ["arms", "core", "back"],
        clonedFromIds: [],
        createdAt: new Date(2021, 2, 2),
        creator: {
          id: "user1",
          name: "Test-User",
          avatar: "https://robohash.org/test-user",
        },
        exercises: [
          {
            id: "1",
            name: "Exercise 1",
            description: "Test d",
            duration: 10,
            repetitions: 0,
          },
          {
            id: "2",
            name: "Exercise 2",
            description: "Test d2",
            duration: 20,
            repetitions: 0,
          },
          {
            id: "3",
            name: "Exercise 3",
            description: "Test d3",
            duration: 0,
            repetitions: 30,
          },
        ],
      }}
      actions={{
        toggleStartStop: () => {},
        togglePause: () => {},
        previousExercise: () => {},
        nextExercise: () => {},
        goToWorkouts: () => {},
      }}
    />
  );

  expect(workout).toMatchSnapshot();
});
