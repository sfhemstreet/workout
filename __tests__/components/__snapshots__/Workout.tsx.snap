// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workout renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-dsxet2-Container e1jr9ft612"
      >
        <div
          class="css-1msu9jd-Top e1jr9ft611"
        >
          <div
            class="css-jkwt5r-Title e1jr9ft610"
          >
            Workout:
            <h4
              class="css-13j9mg8-H4-TextMixin e1uetepz0"
            >
              Workout 1
            </h4>
          </div>
          <section
            class="css-14vvsgy-Container e2o2v6w0"
          >
            Round
            <span>
              2
            </span>
            of 
            3
          </section>
        </div>
        <div
          class="css-jrlue4-FadableDiv-FadeInOutMixin-Exercise e1jr9ft69"
        >
          <div
            class="css-y4beyk-Container erpp7bl2"
          >
            <button
              aria-label="View Exercise Description"
              class="css-1btwv9f-TertiaryButton-SurfaceElevation-TitleButton erpp7bl1"
              data-microtip-position="bottom"
              role="tooltip"
            >
              <h2
                class="css-1vcqgq1-H2-TextMixin ewijl6f0"
              >
                Exercise 1
              </h2>
            </button>
            <div
              class="css-l6wh3n-Container-SurfaceElevation e1nx9dx85"
            >
              <div
                class="css-124tvsf-Paused e1nx9dx80"
              >
                <h1
                  class="css-1891q7e-H1-TextMixin e1ez90bh0"
                >
                  PAUSED
                </h1>
              </div>
              <h1
                class="css-1gwmhuz-Time e1nx9dx84"
              >
                5
              </h1>
              <svg
                class="css-hbuico-SVG e1nx9dx83"
              >
                <defs>
                  <lineargradient
                    id="awesome-counter-gradient"
                    spreadMethod="pad"
                    x1="0%"
                    x2="0%"
                    y1="0%"
                    y2="100%"
                  >
                    <stop
                      offset="0%"
                      stop-color="rgb(187, 134, 252)"
                    />
                    <stop
                      offset="51%"
                      stop-color="rgb(3, 218, 198)"
                    />
                    <stop
                      offset="100%"
                      stop-color="rgb(55, 0, 179)"
                    />
                  </lineargradient>
                </defs>
                <circle
                  class="css-kwiswx-CircleNoAnimation-Circle e1nx9dx81"
                />
              </svg>
            </div>
            <div
              class="css-1i0uwid-SpaceMaker erpp7bl0"
            >
              <div
                class="css-jnb7cp-Container-FadeInOutMixin eolhi291"
              >
                <p
                  class="css-44y8rz-P-TextMixin elrci8c0"
                >
                  Next Up: 
                  Exercise 2
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-508ke0-FadableDiv-FadeInOutMixin-List e1jr9ft68"
        >
          <div
            class="css-16xm46a-Container-SurfaceElevation es7f79g3"
          >
            <div
              class="css-f6kk0r-Title es7f79g0"
            >
              <h5>
                Exercise
              </h5>
              <h5>
                Duration/Reps
              </h5>
            </div>
            <ol
              class="css-o2mk7l-List-SurfaceElevation es7f79g2"
            >
              <li
                class="css-14xr9ka-Item es7f79g1"
              >
                <span>
                  Exercise 1
                  :
                </span>
                <span
                  class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
                >
                  0:10
                </span>
              </li>
              <li
                class="css-840qmy-Item es7f79g1"
              >
                <span>
                  Exercise 2
                  :
                </span>
                <span
                  class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
                >
                  0:20
                </span>
              </li>
              <li
                class="css-840qmy-Item es7f79g1"
              >
                <span>
                  Exercise 3
                  :
                </span>
                <span
                  class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
                >
                  30
                </span>
              </li>
            </ol>
          </div>
        </div>
        <ul
          class="css-c9jeew-ActionsPanel e1jr9ft66"
        >
          <li
            class="css-l6sgid-PreviousArea-FadeInOutMixin e1jr9ft64"
          >
            <button
              class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
            >
              Back
            </button>
          </li>
          <li
            class="css-nsp0rf-NextArea-FadeInOutMixin e1jr9ft62"
          >
            <button
              class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
            >
              Next
            </button>
          </li>
          <li
            class="css-o2jgbv-PauseArea-FadeInOutMixin e1jr9ft63"
          >
            <button
              class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
            >
              Pause
            </button>
          </li>
          <li
            class="css-11q3x0g-StartQuitArea e1jr9ft65"
          >
            <button
              class="css-1gam594-SecondaryButton-SurfaceElevation-StartQuitBtn e1jr9ft60"
            >
              Quit
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dsxet2-Container e1jr9ft612"
    >
      <div
        class="css-1msu9jd-Top e1jr9ft611"
      >
        <div
          class="css-jkwt5r-Title e1jr9ft610"
        >
          Workout:
          <h4
            class="css-13j9mg8-H4-TextMixin e1uetepz0"
          >
            Workout 1
          </h4>
        </div>
        <section
          class="css-14vvsgy-Container e2o2v6w0"
        >
          Round
          <span>
            2
          </span>
          of 
          3
        </section>
      </div>
      <div
        class="css-jrlue4-FadableDiv-FadeInOutMixin-Exercise e1jr9ft69"
      >
        <div
          class="css-y4beyk-Container erpp7bl2"
        >
          <button
            aria-label="View Exercise Description"
            class="css-1btwv9f-TertiaryButton-SurfaceElevation-TitleButton erpp7bl1"
            data-microtip-position="bottom"
            role="tooltip"
          >
            <h2
              class="css-1vcqgq1-H2-TextMixin ewijl6f0"
            >
              Exercise 1
            </h2>
          </button>
          <div
            class="css-l6wh3n-Container-SurfaceElevation e1nx9dx85"
          >
            <div
              class="css-124tvsf-Paused e1nx9dx80"
            >
              <h1
                class="css-1891q7e-H1-TextMixin e1ez90bh0"
              >
                PAUSED
              </h1>
            </div>
            <h1
              class="css-1gwmhuz-Time e1nx9dx84"
            >
              5
            </h1>
            <svg
              class="css-hbuico-SVG e1nx9dx83"
            >
              <defs>
                <lineargradient
                  id="awesome-counter-gradient"
                  spreadMethod="pad"
                  x1="0%"
                  x2="0%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    stop-color="rgb(187, 134, 252)"
                  />
                  <stop
                    offset="51%"
                    stop-color="rgb(3, 218, 198)"
                  />
                  <stop
                    offset="100%"
                    stop-color="rgb(55, 0, 179)"
                  />
                </lineargradient>
              </defs>
              <circle
                class="css-kwiswx-CircleNoAnimation-Circle e1nx9dx81"
              />
            </svg>
          </div>
          <div
            class="css-1i0uwid-SpaceMaker erpp7bl0"
          >
            <div
              class="css-jnb7cp-Container-FadeInOutMixin eolhi291"
            >
              <p
                class="css-44y8rz-P-TextMixin elrci8c0"
              >
                Next Up: 
                Exercise 2
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-508ke0-FadableDiv-FadeInOutMixin-List e1jr9ft68"
      >
        <div
          class="css-16xm46a-Container-SurfaceElevation es7f79g3"
        >
          <div
            class="css-f6kk0r-Title es7f79g0"
          >
            <h5>
              Exercise
            </h5>
            <h5>
              Duration/Reps
            </h5>
          </div>
          <ol
            class="css-o2mk7l-List-SurfaceElevation es7f79g2"
          >
            <li
              class="css-14xr9ka-Item es7f79g1"
            >
              <span>
                Exercise 1
                :
              </span>
              <span
                class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
              >
                0:10
              </span>
            </li>
            <li
              class="css-840qmy-Item es7f79g1"
            >
              <span>
                Exercise 2
                :
              </span>
              <span
                class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
              >
                0:20
              </span>
            </li>
            <li
              class="css-840qmy-Item es7f79g1"
            >
              <span>
                Exercise 3
                :
              </span>
              <span
                class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
              >
                30
              </span>
            </li>
          </ol>
        </div>
      </div>
      <ul
        class="css-c9jeew-ActionsPanel e1jr9ft66"
      >
        <li
          class="css-l6sgid-PreviousArea-FadeInOutMixin e1jr9ft64"
        >
          <button
            class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
          >
            Back
          </button>
        </li>
        <li
          class="css-nsp0rf-NextArea-FadeInOutMixin e1jr9ft62"
        >
          <button
            class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
          >
            Next
          </button>
        </li>
        <li
          class="css-o2jgbv-PauseArea-FadeInOutMixin e1jr9ft63"
        >
          <button
            class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
          >
            Pause
          </button>
        </li>
        <li
          class="css-11q3x0g-StartQuitArea e1jr9ft65"
        >
          <button
            class="css-1gam594-SecondaryButton-SurfaceElevation-StartQuitBtn e1jr9ft60"
          >
            Quit
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Workout with Confetti renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-dsxet2-Container e1jr9ft612"
      >
        <div
          class="css-1msu9jd-Top e1jr9ft611"
        >
          <div
            class="css-jkwt5r-Title e1jr9ft610"
          >
            Workout:
            <h4
              class="css-13j9mg8-H4-TextMixin e1uetepz0"
            >
              Workout 1
            </h4>
          </div>
          <section
            class="css-14vvsgy-Container e2o2v6w0"
          >
            Round
            <span>
              2
            </span>
            of 
            3
          </section>
        </div>
        <div
          class="css-12r5frn-Completed e1jr9ft67"
        >
          <h3
            class="css-33v7o8-H3-TextMixin eyl1k3n0"
          >
            <span
              class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
            >
              WORKOUT COMPLETED
            </span>
          </h3>
          <h3
            class="css-33v7o8-H3-TextMixin eyl1k3n0"
          >
            Congratulations!
          </h3>
        </div>
        <ul
          class="css-c9jeew-ActionsPanel e1jr9ft66"
        >
          <li
            class="css-l6sgid-PreviousArea-FadeInOutMixin e1jr9ft64"
          >
            <button
              class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
            >
              Back
            </button>
          </li>
          <li
            class="css-nsp0rf-NextArea-FadeInOutMixin e1jr9ft62"
          >
            <button
              class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
            >
              Next
            </button>
          </li>
          <li
            class="css-o2jgbv-PauseArea-FadeInOutMixin e1jr9ft63"
          >
            <button
              class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
            >
              Pause
            </button>
          </li>
          <li
            class="css-11q3x0g-StartQuitArea e1jr9ft65"
          >
            <button
              class="css-1gam594-SecondaryButton-SurfaceElevation-StartQuitBtn e1jr9ft60"
            >
              Quit
            </button>
          </li>
        </ul>
        <canvas
          style="position: fixed; pointer-events: none; width: 100%; height: 100%; top: 0px; left: 0px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dsxet2-Container e1jr9ft612"
    >
      <div
        class="css-1msu9jd-Top e1jr9ft611"
      >
        <div
          class="css-jkwt5r-Title e1jr9ft610"
        >
          Workout:
          <h4
            class="css-13j9mg8-H4-TextMixin e1uetepz0"
          >
            Workout 1
          </h4>
        </div>
        <section
          class="css-14vvsgy-Container e2o2v6w0"
        >
          Round
          <span>
            2
          </span>
          of 
          3
        </section>
      </div>
      <div
        class="css-12r5frn-Completed e1jr9ft67"
      >
        <h3
          class="css-33v7o8-H3-TextMixin eyl1k3n0"
        >
          <span
            class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
          >
            WORKOUT COMPLETED
          </span>
        </h3>
        <h3
          class="css-33v7o8-H3-TextMixin eyl1k3n0"
        >
          Congratulations!
        </h3>
      </div>
      <ul
        class="css-c9jeew-ActionsPanel e1jr9ft66"
      >
        <li
          class="css-l6sgid-PreviousArea-FadeInOutMixin e1jr9ft64"
        >
          <button
            class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
          >
            Back
          </button>
        </li>
        <li
          class="css-nsp0rf-NextArea-FadeInOutMixin e1jr9ft62"
        >
          <button
            class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
          >
            Next
          </button>
        </li>
        <li
          class="css-o2jgbv-PauseArea-FadeInOutMixin e1jr9ft63"
        >
          <button
            class="css-ak8ya3-SecondaryButton-SurfaceElevation e1dg5x0l0"
          >
            Pause
          </button>
        </li>
        <li
          class="css-11q3x0g-StartQuitArea e1jr9ft65"
        >
          <button
            class="css-1gam594-SecondaryButton-SurfaceElevation-StartQuitBtn e1jr9ft60"
          >
            Quit
          </button>
        </li>
      </ul>
      <canvas
        style="position: fixed; pointer-events: none; width: 100%; height: 100%; top: 0px; left: 0px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
